### PR TITLE
build hunspell-sys also on Windows

### PR DESCRIPTION
**Pull request by msuesskraut**
- msvc does not support autotools &
  hunspell's build scripts itself do not play well with msvc
- just compile the hunspell-1.7.lib with cc "by hand"
- keep build without msvc as it is

### CODE REVIEW

**Comment by drahnr** [comment]
Very nice work!

**Comment by msuesskraut** [comment]
> 
> 
> It could also be a fallback if autotools is not installed.

It would be nice to autodetect, if autotools are available, but the autotools crate does not provide a check. It plainly assumes, that autotools are on `PATH`.

**Comment by msuesskraut** [comment]
I hid the build with `cc` behind a feature `build-cc`. Default is still to use autotools (with the new `build-autotools` feature). I added a Building section into `Readme.md` to explain this.

Cargo seems not to allow target specific default features. However, dependent crates can specify targets specific features for `hunspell-sys`.

```toml
[target.'cfg(target_env = "msvc")'.dependencies]
hunspell-sys = { version = "0.1.3", default-features = false, features = ["build-cc"] }

[target.'cfg(not(target_env = "msvc"))'.dependencies]
hunspell-sys = "0.1.3"
```

**Comment by drahnr** [comment]
@euclio anything left to do here? This is blocking [cargo-spellcheck](https://github.com/drahnr/cargo-spellcheck) from being installed on windows, so I'd see this merged rather sooner than later :)

**Comment by euclio** [comment]
I don't think a feature flag is the right solution here. I would prefer either using `cc` as a fallback if autotools is not installed, or switching the build to use `cc` entirely.

**Comment by drahnr** [comment]
Using `cc` and subsequent static linkage would cause some license friction. Frankly right now I am not sure if this is even the right place to impl the feature flag, or if it might be better suited for `hunspell-rs` rather than the generated bindings.
What speaks against a feature flag?

**Comment by euclio** [comment]
[Cargo features are not mutually exclusive](https://github.com/rust-lang/cargo/issues/2980). If I try to build this PR with `cargo build --all-features` it will not work.

I'm unfamiliar with the linking impact on licensing here. Why is statically linking not an issue for MSVC but it is for other platforms?

**Comment by drahnr** [comment]
It's not about msvc, it's about static linkage with `libhunspell`

**Comment by euclio** [comment]
We can use `libhunspell` under the MPL license, which has no linking restrictions. In my understanding, since we haven't modified the hunspell source, we're free to statically link and distribute it as long as it's clear where to find the source.

MPL FAQ: https://www.mozilla.org/en-US/MPL/2.0/FAQ/

**Comment by drahnr** [comment]
Huh, mea culpa, I had something GPL-ish in the back of my head. Discard my previous comments then.

**Comment by euclio** [comment]
Ping @msuesskraut, any interest in finishing this?

**Comment by msuesskraut** [comment]
The question is on how to rewrite it. We could orient ourselves on `libsqlite3-sys`:
- `libsqlite3-sys` takes the approach to search for an installed sqlite library (e.g. via `pkg-config``) by default
- optionally: by the feature `bundled` a bundled library is compiled with `cc`
- details: https://github.com/rusqlite/rusqlite#user-content-notes-on-building-rusqlite-and-libsqlite3-sys

Hence, that would mean to search for the hunspell library by default via `pkg-config`.

However, with the bundled feature the hunspell library will be compiled from source by `cc`.

Is this the way to move forward? If yes, I could adapt my pull request.

**Comment by drahnr** [comment]
I would very much appreciate that approach.

**Comment by euclio** [comment]
Yeah, that sounds great to me.

**Comment by msuesskraut** [comment]
So I updated the pull request as discussed. Sorry for the many commits. I'm not so fluent with travis. I copied the approach to test the `bundled` feature from `rusqlite`.

I also updated the build dependencies to the most current versions.

Please have a look.

File: README.md [hunk 1]
```diff
@@ -7,3 +7,15 @@
 Rust bindings for the [hunspell] C API.
 
 [hunspell]: https://hunspell.github.io/
+
+## Building
+
+By default `hunspell-sys` searches while building for hunspell library with `pkg-config`.
```
Comment (euclio) [inline README.md]:
```suggestion
By default `hunspell-sys` searches for a hunspell library installation with `pkg-config`.
```

**Review by euclio — COMMENTED** [review]
Looks great! Just one suggestion.

Don't worry too much about travis, I'll have to change it anyways to get it to work with Windows. (I'll probably migrate to GHA too).

**Comment by euclio** [comment]
Published v0.2.0.

**Comment by drahnr** [comment]
Thank you!

**Comment by msuesskraut** [comment]
Thanks!

**Comment by drahnr** [comment]
Just for sake of completeness:

https://hunspell.github.io/ states the following regarding licensing:

> C++ library under GPL/LGPL/MPL tri-license.

**Pull request merged**